### PR TITLE
stronger dependsOn check for features & apps 

### PR DESCRIPTION
**Pull request by dshimo**
fix case for when server has features and no apps
assumption that if no apps defined, uses war plugin

### CODE REVIEW

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy [hunk 1]
```diff
@@ -190,15 +185,7 @@ class Liberty implements Plugin<Project> {
...
-            dependsOn project.tasks.withType(War)
-
-            project.afterEvaluate {
-                if (server.features.name != null && !server.features.name.isEmpty()) {
-                    dependsOn 'installFeature'
-                } else {
-                    dependsOn 'libertyCreate'
-                }
-           }
+            dependsOn project.tasks.withType(War), 'libertyCreate'
```
Comment (cthigh) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: remove for uninstallFeature

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy [hunk 2]
```diff
@@ -51,9 +51,8 @@ class Liberty implements Plugin<Project> {
 
         //Create expected server extension from liberty extension data
         project.afterEvaluate{
-            if (project.liberty.server == null) {
-                project.liberty.server = copyProperties(project.liberty)
-            }
+            project.liberty.server = (project.liberty.server) ?: copyProperties(project.liberty)
```
Comment (cthigh) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: ?:

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy [hunk 3]
```diff
@@ -262,17 +249,29 @@ class Liberty implements Plugin<Project> {
...
+
+        if (apps) tasks.add('installApps')
+        if (feature) tasks.add('installFeature')
+        if (!apps && !feature) tasks.add(elseDepends)
+        return tasks
+    }
+
+    private boolean dependsOnApps(ServerExtension server) {
+        return ((server.apps != null || server.dropins != null) ||
+                (server.bootstrapProperties != null && server.bootstrapProperties.containsKey('testAppLocation')))
```
Comment (cthigh) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]:
```
        if ((server.apps == null || server.apps.isEmpty()) && (server.dropins == null || server.dropins.isEmpty())) {
            if (project.plugins.hasPlugin('war')) {
                server.apps = [project.war]
            }
}
```

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy (hunk 1 above)
Comment (dshimo) [reply src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: uninstallFeature is clean. leaving this line under installApps

Comment (cthigh) [reply src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: I think that the code must have been collapsed and I thought it was under uninstall.

**Pull request merged**